File: 2022/D1.hs
import System.IO

main :: IO ()
main = do
  -- Open the input file
  inputFile <- openFile "input.txt" ReadMode
  -- Read the input file line by line
  inputLines <- hGetContents inputFile
  let
    -- Convert each line to an integer and add it to a list of calories
    -- If the line is empty, create a new list for the next Elf's food items
    calories = map (map read . lines) . groupBy (\x y -> x /= "") . lines $ inputLines
  -- Find the sum of the maximum calorie list
  let maxCalories = maximum $ map sum calories
  -- Print the result
  print maxCalories
  -- Close the input file
  hClose inputFile

File: 2022/D1.rb
# Get the input file name from the command line argument
input_file = ARGV[0]

# Read the input file line by line
calories = File.readlines(input_file).map(&:to_i)

# Group the calories by Elf
calories_by_elf = calories.chunk { |c| c.zero? }.map(&:last)

# Find the sum of the maximum calorie list
max_calories = calories_by_elf.max_by(&:sum).sum

# Print the result
puts max_calories
